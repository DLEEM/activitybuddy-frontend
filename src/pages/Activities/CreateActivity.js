import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { createActivity } from '../../services/clientToBackend';
import { Form, FormGroup, Col, FormControl, Button, ControlLabel } from 'react-bootstrap'

class CreateActivity extends Component {
  constructor(props) {
    super(props)
    this.state = {
      successActivity: false,
      activity: {
        name: ""
      }
    }
  }

  render() {
    return (
      <div>
        <Form horizontal onSubmit={this.onSubmit}>
          <FormGroup controlId="formHorizontalName">
            <Col componentClass={ControlLabel}>
              Name of Activity
            </Col>
            <Col>
              <FormControl onChange={this.onChange} name="name" type="text" value={this.state.activity.name} placeholder="Name of Activity" />
            </Col>
          </FormGroup>

          <FormGroup>
            <Col>
              <Button type="submit">
                Create Activity
              </Button>
            </Col>
          </FormGroup>
        </Form>
        {this.state.successActivity && <Redirect to="/activities" /> }
      </div>
    );
  }

  onChange = (e) => {
    let {activity} = this.state
    activity[e.target.name] = e.target.value
    this.setState ({ activity })
  }

  onSubmit = (e) => {
    e.preventDefault()
    createActivity(this.state.activity)
    .then(json => {
      console.log("got to the second then:", json)
      if (json.errors) {
        console.log("ERRORS", json.errors)
        this.setState({ errors: json.errors })
      } else {
        console.log("i am else")
        this.setState ({ successActivity: true })
      }
    })
  }
}

export default CreateActivity;
