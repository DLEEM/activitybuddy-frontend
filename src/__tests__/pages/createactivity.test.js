import React from 'react'
import CreateActivity from '../../pages/Activities/CreateActivity'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

const newActivity = { name: "Drawing" }

test('renders the create activity correctly', () => {
  const json = renderer
    .create(<CreateActivity activity={newActivity} /> )
    .toJSON()
  expect(json).toMatchSnapshot()
})

test('the title should display if it is a state', () => {
  const wrapper = shallow(<CreateActivity activity={newActivity} />)
  const activityShallow = wrapper.find('.ActivityNew')
  expect(activityShallow.find('.title')).toBeDefined()
})

// console.log('should have title', activitiyShallow.debug())
