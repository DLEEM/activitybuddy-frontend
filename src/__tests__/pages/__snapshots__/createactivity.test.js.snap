// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the create activity correctly 1`] = `
<div
  className="ActivityNew"
>
  <h2
    class="title"
  >
    title
  </h2>
  <form
    className="form-horizontal"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="formHorizontalName"
      >
        Name of Activity
      </label>
      <div
        className=""
      >
        <input
          className="form-control"
          id="formHorizontalName"
          name="name"
          onChange={[Function]}
          placeholder="Name of Activity"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className=""
      >
        <button
          className="btn btn-default"
          disabled={false}
          type="submit"
        >
          Create Activity
        </button>
      </div>
    </div>
  </form>
</div>
`;
